### PR TITLE
Add tests for sycl::kernel to cover remaining gaps

### PR DESCRIPTION
**Pull request by vladimirkhashev**
Add tests for sycl::kernel to cover remaining gaps according to test plan https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/kernel.asciidoc.

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

**Comment by fraggamuffin** [comment]
LGTM

**Review by keryell — APPROVED** [review]
Thanks.

File: tests/kernel/kernel_constructors.cpp [hunk 1]
```diff
@@ -21,276 +21,219 @@
...
-          }
-        }
+  if (ctsQueue.get_backend() == sycl::backend::opencl) {
+    auto iopKernelA = sycl::get_native<sycl::backend::opencl>(kernelA);
+    auto iopKernelB = sycl::get_native<sycl::backend::opencl>(kernelB);
+    if (!ctsSelector.is_host() && (iopKernelA != iopKernelB)) {
+      FAIL(
+          "kernel was not constructed correctly. (contains different "
+          "OpenCL kernel object)");
+    }
```
Comment (bader) [inline tests/kernel/kernel_constructors.cpp:56]: Should we use `CHECK` instead of `if` + `FAIL`?

Comment (vladimirkhashev) [reply tests/kernel/kernel_constructors.cpp]: Changed it.

Comment (bader) [reply tests/kernel/kernel_constructors.cpp]: Thanks!

**Pull request merged**